Properly translate Theia variables to AI variables where possible

martin-fleck-at opened the pull request:
<!--
Thank you for your Pull Request. Please provide a description and review
the requirements below.

Contributors guide: https://github.com/theia-ide/theia/blob/master/CONTRIBUTING.md

Note: Security vulnerabilities should not be disclosed on GitHub, through a PR or any
other means. See SECURITY.md at the root of this repository, to learn how to report
vulnerabilities.
-->

#### What it does

- Move today variable contribution to common where it should have been

Relates to https://github.com/eclipsesource/osweek-2024/issues/46

#### How to test

- Try variables such as '#file', '#execPath', '#workspaceRoot' or something similar
- Try also variables with arguments such as '#env:PATH'

#### Follow-ups

<!-- Please list potential follow-up work, including known issues, possible future work, identified technical debt, and potentially introduced technical debt. If the PR introduces technical debt, specify the reason why this is acceptable. Please create tickets and link them here. Please use the label "technical debt" for new issues when it applies. -->

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

lucas-koehler reviewed (approved):
LGTM thanks :sparkles: Please fix the conflicts :D

martin-fleck-at commented:
@lucas-koehler Thank you, I rebased the change on the current master to fix the conflicts.
@JonasHelming I introduced the `currentText` variable to get the complete content of the currently opened text editor
